src/eslint/__tests__/react-server-components-directive.test.js

```diff
@@ -0,0 +1,139 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { describe, test } from "vitest";
+import { RuleTester } from "eslint";
+import rscDirective from "../react-server-components-directive.js";
+
+const testRule = testCase => {
+  new RuleTester({ parserOptions: { ecmaVersion: 2018, sourceType: "module" } }).run(
+    "react-server-components-directive",
+    rscDirective,
+    { valid: [], invalid: [], ...testCase },
+  );
```

Comment:
`RuleTester` utility relies on globals `describe`, `it` to be present. They do not exist in vitest, so we need to build a custom utility to run rule tester in manually declared test-cases